Sources/SourceControl/Repository.swift

```diff
@@ -42,7 +42,7 @@ public struct RepositorySpecifier: Hashable {
 
     /// Returns the cleaned basename for the specifier.
     public var basename: String {
-        var basename = self.location.description.components(separatedBy: "/").last(where: { !$0.isEmpty }) ?? ""
+        var basename = self.url.pathComponents.last(where: { !$0.isEmpty })!
```

Comment:
cleanup looks good, but lets not force unwarp

Reply:
I think that the empty string is wrong - a repository _should_ have a base name even if it is bare.  Is there some case that it would be empty?  Converting it to a throwing function makes sense.

Reply:
Wait, this is a computed property.  We can make it into an effectful property, but that might have a large fallout.

Reply:
maybe that is why it was defaulted to empty? perhaps one way to deal with this is to default to empty by add an assert to that at least in development time we know something bad has happened?

Reply:
Hmm I went with a slightly different approach - `guard` + `fatalError` ... though I suppose the result is roughly the same.